hadoop-crypto/src/main/java/com/palantir/hadoop/PathConvertingFileSystem.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.hadoop;
+
+import com.google.common.base.Function;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.net.URI;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.permission.FsPermission;
+import org.apache.hadoop.util.Progressable;
+
+/**
+ * A decorator {@link FileSystem} that delegates calls and converts paths to/from the delegate FileSystem.
+ * {@link Path}s passed into methods are converted using {@code toDelegatePathFunc} before being forwarded.
+ * {@link Path}s returned from the delegate are converted using {@code toReturnPathFunc} before being returned.
+ */
+public final class PathConvertingFileSystem extends FileSystem {
+
+    private final FileSystem delegate;
+    private final Function<Path, Path> toDelegatePathFunc;
```

Comment:
changed the function parameters to `toFunc` and `fromFunc` and then my helper methods to `to` and `from`.